adv-rumors.lua

```diff
@@ -1,83 +1,127 @@
--- Improve "Bring up specific incident or rumor" menu in Adventure mode
---@ module = true
---[====[
-
-adv-rumors
-==========
-Improves the "Bring up specific incident or rumor" menu in Adventure mode.
-
-- Moves entries into one line
-- Adds a "slew" keyword for filtering, making it easy to find your kills and not your companions'
-- Trims repetitive words
-
-]====]
-
---========================
--- Author : 1337G4mer on bay12 and reddit
--- Version : 0.2
--- Description : A small utility based on dfhack to improve the rumor UI in adventure mode.
---
---      In game when you want to boast about your kill to someone. Start conversation and choose
---      the menu "Bring up specific incident or rumor"
---      type rumors in dfhack window and hit enter. Or do the below keybind and use that directly from DF window.
---
--- Prior Configuration: (you can skip this if you want)
---      Set the three boolean values below and play around with the script as to how you like
---      improveReadability = will move everything in one line
---      addKeywordSlew = will add a keyword for filtering using slew, making it easy to find your kills and not your companion's
---      shortenString = will further shorten the line to = slew "XYZ" ( "n time" ago in " Region")
---=======================
-
-local utils = require "utils"
-
-local names_blacklist = utils.invert{"a", "an", "you", "attacked", "slew", "was", "slain", "by"}
-
-function condenseChoiceTitle(choice)
-    while #choice.title > 1 do
-        choice.title[0].value = choice.title[0].value .. ' ' .. choice.title[1].value
-        choice.title:erase(1)
+-- Improve "Bring up specific incident or rumor", "Ask for Directions" and "Ask about Somebody" menus in Adventure mode
+
+--@ module=true
+
+-- requirements
+local overlay = require('plugins.overlay')
+local utils = require('utils')
+local widgets = require('gui.widgets')
+
+-- globals
+ignore_words = utils.invert{
+    "a", "an", "attacked", "by", "in", "occurred", "of", "or",
+    "s", "slain", "slew", "the", "this", "to", "was", "which"
+}
+
+-- locals
+local adventure = df.global.game.main_interface.adventure
+
+AdvRumorsOverlay = defclass(AdvRumorsOverlay, overlay.OverlayWidget)
+AdvRumorsOverlay.ATTRS{
+    name='adv-rumors',
+    desc='Adds keywords to conversation entries.',
+    overlay_only=true,
+    default_enabled=true,
+    viewscreens='dungeonmode/Conversation',
+}
+
+function AdvRumorsOverlay:render()
+    rumorUpdate()
+end
+
+OVERLAY_WIDGETS = {message=AdvRumorsOverlay}
```

Comment:
"message" isn't quite a good descriptive name for this overlay. how about `keyworder` or `conversation`?